Rest tests intermittently fail in CI

### Description

The Rest tests sometimes fail in CI, when reran they succeed. It seems that the tests have become flaky because of the recent change to have a docker db container for each Jest worker. See https://github.com/hashgraph/hedera-mirror-node/issues/7561

### Steps to reproduce

1. See rest tests in CI. 

### Additional context

```
> Task :rest:test
(Use `node --trace-warnings ...` to show where the warning was created)
(node:2942) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
PASS __tests__/utils.test.js (4696 ms)
(node:2943) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
PASS __tests__/controllers/contractController.test.js (5643 ms)
PASS __tests__/tokens.test.js (1417 ms)
PASS __tests__/utilsFilters.test.js (1187 ms)
PASS __tests__/transactions.test.js (1116 ms)
PASS __tests__/controllers/accountController.test.js (1540 ms)
ERROR: Unable to obtain connection from database (jdbc:postgresql://localhost:[32](https://github.com/hashgraph/hedera-mirror-node/actions/runs/7802341291/job/21279568149#step:5:33)770/mirror_node) for user 'mirror_node': Connection to localhost:32770 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Connection to localhost:32770 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

Caused by: org.postgresql.util.PSQLException: Connection to localhost:32770 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
Caused by: java.net.ConnectException: Connection refused
```

### Hedera network

other

### Version

main

### Operating system

None

--- a/hedera-mirror-rest/__tests__/globalSetup.js
+++ b/hedera-mirror-rest/__tests__/globalSetup.js
@@ -42,7 +42,12 @@ const createDbContainers = async (maxWorkers) => {
       .withDatabase(dbName)
       .withPassword('mirror_node_pass')
       .withUsername('mirror_node')
-      .withWaitStrategy(Wait.forHealthCheck())
+      .withLogConsumer((stream) => {
+        stream.on('data', (line) => console.log(line));
+        stream.on('err', (line) => console.error(line));
+        stream.on('end', () => console.log('Stream closed'));
+      })
+      .withWaitStrategy(Wait.forLogMessage('database system is ready to accept connections'))
       .start();
     console.info(`Started PostgreSQL container for Jest worker ${i} with image ${image}`);
     setJestEnvironment(dockerDb, i);

--- a/hedera-mirror-rest/__tests__/globalSetup.js
+++ b/hedera-mirror-rest/__tests__/globalSetup.js
@@ -42,7 +42,6 @@ const createDbContainers = async (maxWorkers) => {
       .withDatabase(dbName)
       .withPassword('mirror_node_pass')
       .withUsername('mirror_node')
-      .withStartupTimeout(100)
       .withWaitStrategy(Wait.forHealthCheck())
       .start();
     console.info(`Started PostgreSQL container for Jest worker ${i} with image ${image}`);

--- a/hedera-mirror-rest/__tests__/globalSetup.js
+++ b/hedera-mirror-rest/__tests__/globalSetup.js
@@ -47,7 +47,7 @@ const createDbContainers = async (maxWorkers) => {
         stream.on('err', (line) => console.error(line));
         stream.on('end', () => console.log('Stream closed'));
       })
-      .withWaitStrategy(Wait.forLogMessage('database system is ready to accept connections'))
+      .withWaitStrategy(Wait.forLogMessage('database system is ready to accept connections', 2))
       .start();
     console.info(`Started PostgreSQL container for Jest worker ${i} with image ${image}`);
     setJestEnvironment(dockerDb, i);

--- a/hedera-mirror-rest/__tests__/globalSetup.js
+++ b/hedera-mirror-rest/__tests__/globalSetup.js
@@ -18,6 +18,7 @@ import fs from 'fs';
 import os from 'os';
 import path from 'path';
 import {PostgreSqlContainer} from '@testcontainers/postgresql';
+import {Wait} from 'testcontainers';
 
 const dbName = 'mirror_node';
 const dockerNamePrefix = 'INTEGRATION_DATABASE_URL';
@@ -41,6 +42,8 @@ const createDbContainers = async (maxWorkers) => {
       .withDatabase(dbName)
       .withPassword('mirror_node_pass')
       .withUsername('mirror_node')
+      .withStartupTimeout(100)
+      .withWaitStrategy(Wait.forHealthCheck())
       .start();
     console.info(`Started PostgreSQL container for Jest worker ${i} with image ${image}`);
     setJestEnvironment(dockerDb, i);

--- a/hedera-mirror-rest/__tests__/globalSetup.js
+++ b/hedera-mirror-rest/__tests__/globalSetup.js
@@ -42,11 +42,6 @@ const createDbContainers = async (maxWorkers) => {
       .withDatabase(dbName)
       .withPassword('mirror_node_pass')
       .withUsername('mirror_node')
-      .withLogConsumer((stream) => {
-        stream.on('data', (line) => console.log(line));
-        stream.on('err', (line) => console.error(line));
-        stream.on('end', () => console.log('Stream closed'));
-      })
       .withWaitStrategy(Wait.forLogMessage('database system is ready to accept connections', 2))
       .start();
     console.info(`Started PostgreSQL container for Jest worker ${i} with image ${image}`);

